vllm/v1/core/scheduler.py

```diff
@@ -642,14 +654,20 @@ def finish_requests(
             request.status = finished_status
             self._free_request(request)
 
+    def _free_prompt_logprobs_cache(self, request_id: str) -> None:
+        if request_id in self._cached_prompt_logprobs:
+            del self._cached_prompt_logprobs[request_id]
+
     def _free_request(self, request: Request) -> None:
         assert request.is_finished()
+        request_id = request.request_id
```

Comment:
Personally, I'd keep refactoring like this out of the PR - e.g. in this case, it would mean that reviewers are just looking at the addition of the `free_prompt_log_probs_cache()` call, not checking the other refactoring changes for correctness

(An alternative approach is to put the refactoring in a separate commit in the same PR ... but in the end, I guess reviewers on this project aren't looking at individual commits)